packages/nextjs-router/src/common/refine-link.tsx

```diff
@@ -11,17 +11,18 @@ type RefineLinkProps = (
     | LinkProps
 ) & {
     children: React.ReactNode;
+    ref?: React.Ref<HTMLAnchorElement>;
 };
 
-export const RefineLink: React.FC<RefineLinkProps> = ({
-    children,
-    ...props
-}) => (
-    <Link
-        href={"to" in props ? props.to : props.href}
-        legacyBehavior={false}
-        {...props}
-    >
-        {children}
-    </Link>
+export const RefineLink: React.FC<RefineLinkProps> = React.forwardRef(
+    ({ children, ...props }) => (
+        <Link
+            href={"to" in props ? props.to : props.href}
```

Comment:
```suggestion
    ({ children, ...props }, ref) => (
        <Link
            ref={ref}
            href={"to" in props ? props.to : props.href}
```

Since we do `forwardRef`, I think it's better to actually do that 🤣

Thank you for the PR! 🙏